src/mmw/apps/geoprocessing_api/views.py

```diff
@@ -576,6 +576,163 @@ def start_analyze_soil(request, format=None):
+                    },
+                    {
+                        "order": 6,
+                        "lengthkm": 44.51,
+                        "slopepct": 0,
+                        "ag_length": 0,
+                        "non_ag_length": 0,
+                    },
+                    {
+                        "order": 7,
+                        "lengthkm": 0,
+                        "slopepct": 0,
+                        "ag_length": 0,
+                        "non_ag_length": 0,
+                    },
+                    {
+                        "order": 8,
+                        "lengthkm": 0,
+                        "slopepct": 0,
+                        "ag_length": 0,
+                        "non_ag_length": 0,
+                    },
+                    {
+                        "order": 9,
+                        "lengthkm": 0,
+                        "slopepct": 0,
+                        "ag_length": 0,
+                        "non_ag_length": 0,
+                    },
+                    {
+                        "order": 10,
+                        "lengthkm": 0,
+                        "slopepct": 0,
+                        "ag_length": 0,
+                        "non_ag_length": 0,
+                    },
+                    {
+                        "order": 999,
+                        "lengthkm": 0,
+                        "slopepct": 0,
+                        "ag_length": 0,
+                        "non_ag_length": 0,
+                    }
+                ]
+            }
+        }
+
+    </details>
+
+    ---
+    type:
+      job:
+        required: true
+        type: string
+      status:
+        required: true
+        type: string
+
+    omit_serializer: true
+    parameters:
+       - name: body
+         description: A valid single-ringed Multipolygon GeoJSON
+                      representation of the shape to analyze.
+                      See the GeoJSON spec
+                      https://tools.ietf.org/html/rfc7946#section-3.1.7
+         paramType: body
+         type: object
+       - name: wkaoi
+         paramType: query
+         description: The table and ID for a well-known area of interest,
+                      such as a HUC.
+                      Format "table__id", eg. "huc12__55174" will analyze
+                      the HUC-12 City of Philadelphia-Schuylkill River.
+
+       - name: Authorization
+         paramType: header
+         description: Format "Token&nbsp;YOUR_API_TOKEN_HERE". When using
+                      Swagger you may wish to set this for all requests via
+                      the field at the top right of the page.
+    consumes:
+        - application/json
+    produces:
+        - application/json
+
+    """
+    user = request.user if request.user.is_authenticated() else None
+    area_of_interest, wkaoi = _parse_input(request)
+
+    return start_celery_job([
+        tasks.analyze_streams.s(area_of_interest)
```

Comment:
I think we can easily add [`nlcd_streams`](https://github.com/WikiWatershed/model-my-watershed/blob/develop/src/mmw/apps/modeling/mapshed/tasks.py#L161) here to calculate agricultural length of streams. I'll create an example.

Reply:
Oh wait, we need agricultural and non-agricultural length _per stream order_. Hmm. Okay let's leave that for a later card.

Reply:
My impression from the original google doc wireframe version was that it was total and not per stream order, but I'll make another card for hashing out those details.

Reply:
My interpretation of the wireframe is the same as kelly's - ag and non-ag lengths are total, not per stream order.

Reply:
Okay. In that case I have an idea on how to do it, but will tackle it in #2528